src/status_im/hardwallet/core.cljs

```diff
@@ -1315,6 +1353,8 @@
                        (assoc-in [:hardwallet :wallet-address] wallet-address)
                        (assoc-in [:hardwallet :encryption-public-key] encryption-public-key)
                        (assoc-in [:hardwallet :keycard-instance-uid] (or instance-uid' instance-uid))
+                       (assoc-in [:hardwallet :keycard-key-uid] key-uid)
+                       (assoc-in [:hardwallet :application-info :key-uid] key-uid)
```

Comment:
why does the information need to be present in two different places?

Reply:
@yenda 
`[:hardwallet :keycard-key-uid]` is used in account creation and will be cleared. 
`[:hardwallet :application-info :key-uid]` is be used for login. Also, it updates on every card tap when we get app-info from keycard.

Reply:
ok I see so they are the same only until next login. What about the one in `:account/account`? Is it `[:hardwallet :keycard-key-uid]` that is getting cleared and placed there afterwards?

Reply:
@yenda yes, `[:hardwallet :keycard-key-uid]` gets stored into `account/account` and gets cleared after

Reply:
could you add a comment there, also why is there so many assoc-in wouldn't it be better at this point to prepare the hardwallet map or merge it with a map containing these changes?

Reply:
good idea, i'm going to put all the data into map and `assoc` it once